src/WebJobs.Extensions.DurableTask/DurabilityProvider.cs

```diff
@@ -418,5 +433,51 @@ internal virtual bool ConnectionNameMatches(DurabilityProvider durabilityProvide
         {
             return this.ConnectionName.Equals(durabilityProvider.ConnectionName);
         }
+
+        /// <summary>
+        /// Returns the instance id of the entity scheduler for a given entity id.
+        /// </summary>
+        /// <param name="entityId">The entity id.</param>
+        /// <returns>The instance id of the scheduler.</returns>
+        protected string GetSchedulerIdFromEntityId(EntityId entityId)
+        {
+            return EntityId.GetSchedulerIdFromEntityId(entityId);
```

Comment:
Is there a reason this is now a `protected` method, as opposed to having people just directly call `EntityId.GetScheduleIdFromEntityId`? If it's because we want to allow each provider to specify their own mapping, then shouldn't this be `virtual`?

Reply:
It is a private method, so the point was to make it "semi-public", meaning accessible to implementors of DurabilityProvider. That way it does not pollute the public API, e.g. normal users won't see it in IntelliSense.